src/main/java/christmas/util/InputValidator.java

```diff
@@ -0,0 +1,103 @@
+package christmas.util;
+
+import christmas.consts.Menu;
+import christmas.consts.SimpleConsts;
+import christmas.consts.Splitter;
+import christmas.domain.MenuOrderAssembler;
+import christmas.exception.DayExceptionCaller;
+import christmas.exception.OrderExceptionCaller;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class InputValidator {
+    private final MenuOrderAssembler menuOrderAssembler;
+
+    public InputValidator(MenuOrderAssembler menuOrderAssembler) {
+        this.menuOrderAssembler = menuOrderAssembler;
+    }
+
+    public int parseToValidatedDate(String inputtedNumber) {
+        int validatedNumber = 0;
+        try {
+            validateEmpty(inputtedNumber);
+            validatedNumber = validateNumber(inputtedNumber);
+        } catch (IllegalArgumentException e) {
+            DayExceptionCaller.throwDayException(e);
+        }
+        validateMemorySafe(validatedNumber);
+        return validatedNumber;
+    }
+
+    private void validateEmpty(String inputtedString) {
+        if (inputtedString == null || inputtedString.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private int validateNumber(String inputtedNumber) {
+        int validatedNumber = 0;
+        try {
+            validatedNumber = Integer.parseInt(inputtedNumber);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(e);
+        }
+        return validatedNumber;
+    }
+
+    private void validateMemorySafe(int validatedNumber) {
+        if (Math.abs(validatedNumber) > SimpleConsts.MEMORY_LIMIT.getValue()) {
+            DayExceptionCaller.throwDayException();
+        }
+    }
+
+    public EnumMap<Menu, Integer> validateMenus(String inputtedDate) {
+        List<String> splitMenuNames = new ArrayList<>();
+        List<Integer> counts = null;
+        try {
+            List<String> splitByComma = splitByRegex(inputtedDate, Splitter.COMMA.getRegex());
+            List<String> splitCounts = new ArrayList<>();
+            splitToMenuAndCounts(splitByComma, splitMenuNames, splitCounts);
+            counts = getValidatedCounts(splitCounts);
+        } catch (IllegalArgumentException e) {
+            OrderExceptionCaller.throwOrderException(e);
```

Comment:
`ExceptionCaller`를 사용해 예외를 발생시키는 로직의 중복을 추출한 부분은 좋아 보입니다!
다만, 이렇게 공통된 예외를 발생시키는 로직이 예외를 "던지도록" 구현을 하는 것에는 한 가지 단점이 있는데요.

`throwOrderException`이라는 메서드가 내부적으로 무조건 예외를 던짐을 우리는 알 수 있지만, `validateMenus` 메서드는 이를 알지 못합니다. 
때문에 아래의 return문이 `try`문 내부에 위치한다면 예외 상황에서의 예외가 정해져 있지 않다는 이유로 컴파일 에러를 발생시킬거에요.

저의 경우는 예외의 중복되는 코드들을 예외를 "만들도록" 추출하고, `catch`문 내부에서는 이렇게 만들어 반환된 예외를 명시적으로 던지도록 구현하는 편입니다.
```suggestion
            throw OrderExceptionCaller.makeOrderException(e);
```

Reply:
저는 일일이 `throw new IllegalArgumentException()`을 던지는 게 보기 안 좋다고 생각을 했습니다.
그래서 이렇게 하나의 `ExceptionCaller`를 만들었는데  예외를 "던지도록" 만드는 게 아닌, "만들도록" 만드는 방법이 있었군요...!!
덕분에 하나 배웁니다 👍